### ISSUE TITLE
[HOLD for payment 2024-01-09] [$250] Migrate FloatingActionButton.js to function component

### ISSUE THREAD

**Issue by marcaaron** [Expensify/App#16150]
### Filenames
- [ ] [src/components/FloatingActionButton.js](https://github.com/Expensify/App/blob/main/src/components/FloatingActionButton.js)
    - **type:** `React.PureComponent`
    - **has state values:** false
    - **has refs:** true
    - **has context:** false
    - **lifecycle methods:** `componentDidUpdate`

### Task
- We currently have some class components in our codebase that we would like to refactor to a function component.
- Here's a link with some general advice on how to refactor a class component to a function component: https://react.dev/reference/react/Component#alternatives
- If you need additional guidance, please ask in [#expensify-open-source](https://expensify.slack.com/archives/C01GTK53T8Q)
- Test for any regressions and verify that there are no breaking changes

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0135528d050de22c41</li>
        <li>Upwork Job ID: 1681842818291113984</li>
        <li>Last Price Increase: 2023-07-20</li>
<li>Automatic offers: </li>
<ul>
<li>rayane-djouah | Contributor | 26487109</li>
</ul></li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
⚠️ Could not update price automatically because there is no linked Upwork Job ID. The BZ team member will need to update the price manually in Upwork.

**Comment by muxriddinmuqimov77** [comment]
I'd like to work on this issue

**Comment by s-alves10** [comment]
I'd love to work on this.

**Comment by dayana7204** [comment]
I am ready to work on this ticket. :)

**Comment by olexyt** [comment]
I can work on this issue.

**Comment by mkhutornyi** [comment]
I'd like to work on this.

**Comment by Skalakid** [comment]
Hey! I’m Michał from Software Mansion, an expert agency, and I’d like to work on this issue.

**Comment by KrAbhas** [comment]
I would like to work on this

**Comment by payal-lathidadiya** [comment]
can i work on this

**Comment by marcaaron** [comment]
@code0406 please read the contributing guidelines and learn to identify issues that need help with the `Help Wanted` label.

**Comment by roryabraham** [comment]
Assigning this to @rayane-djouah per [this discussion](https://expensify.slack.com/archives/C01GTK53T8Q/p1693267309039549). Refactoring `FloatingActionButton` to a functional component appears to be the best way to refactor `Icon` without introducing bugs due to `FloatingActionButton`'s use of `Animated.createAnimatedComponent`, which only works with class components.

**Comment by roryabraham** [comment]
@marcaaron I'm co-assigning myself here since I have context from [slack](https://expensify.slack.com/archives/C01GTK53T8Q/p1693267309039549) and the Icon migration issue. I'll leave you co-assigned in case there's anything from your [WIP PR](https://github.com/Expensify/App/pull/23239) you want to make sure is carried over

**Comment by roryabraham** [comment]
Also going to switch the C+ to @rushatgabhane since he's assigned to [this issue](https://github.com/Expensify/App/issues/16163) and now both issues will be solved with 1 PR.

**Comment by marcaaron** [comment]
I'm gonna unassign myself here. I actually missed your comments and thought that this was still assigned to me and continued to work on the PR. 😬 

FWIW I didn't notice any issues with the FAB when testing, but maybe missed whatever you saw. I will close mine in favor of this one.

Do we anticipate the react-native-reanimated release to come before the EOY?

**Comment by roryabraham** [comment]
Asked in slack: https://expensify.enterprise.slack.com/archives/C04878MDF34/p1701011610023949

**Comment by roryabraham** [comment]
Hopefully Tuesday: https://expensify.enterprise.slack.com/archives/C04878MDF34/p1701026988133139?thread_ts=1701011610.023949&channel=C04878MDF34&message_ts=1701026988.133139

**Comment by roryabraham** [comment]
Looks like the PR was reverted and we have a replacement here: https://github.com/Expensify/App/pull/33336

**Comment by MitchExpensify** [comment]
Not 100% on whether payment is due here seeing as the PR was reverted, mind giving me a BZ buddy check here @mallenexpensify ? Thank you!

**Comment by rushatgabhane** [comment]
@MitchExpensify no payment due yet. 
@rayane-djouah has raised a new PR, let's wait on it's merge for next payment date

**Comment by MitchExpensify** [comment]
Thanks, @rushatgabhane !

**Comment by roryabraham** [comment]
The [new PR](https://github.com/Expensify/App/pull/33336) is currently on staging

**Comment by MitchExpensify** [comment]
Reminder set to pay!

**Payment summary:**

- $250 C+ @rushatgabhane (NewDot)
- $250 C @rayane-djouah (Upwork)

**Comment by MitchExpensify** [comment]
Paid and contract ended @rayane-djouah
